sunshine/video.cpp

```diff
@@ -1215,7 +1226,7 @@ encode_e encode_run_sync(std::vector<std::unique_ptr<sync_session_ctx_t>> &synce
       }
 
       if(pos->img_tmp) {
-        if(pos->hwdevice->convert(*pos->img_tmp)) {
+        if(pos->session.device->convert(*pos->img_tmp)) {
```

Comment:
I wasn't completely sure about this change. I didn't know if you meant to keep that original `hwdevice` around on purpose instead of using `session.device` here.

The problem with the old way is that `make_session()` will create a `swdevice_t` that will handle our scaling in software and copy the YUV data to the GPU:
https://github.com/loki-47-6F-64/sunshine/blob/master/sunshine/video.cpp#L951-L958

With the original code which was just directly storing the result of `disp->make_hwdevice()`, we end up with a instance of the base `hwdevice_t` class which doesn't actually implement `convert()`, `set_frame()`, etc.

Reply:
This code path is supposed te be only reached from Windows nvenc and amdvce since in the "old days" it was using ID3D11VideoDevice to convert image formats on the GPU.

It was simply faster to simply synchronise grabbing the image from the desktop and encoding it.

Reply:
I see. So should I add `SYSTEM_MEMORY` for Linux NVENC to use the async capture path or stick with my existing sync capture approach?

Reply:
Yeah, that's also done for vaapi.

Reply:
OK, I've reverted that portion and added `SYSTEM_MEMORY` instead.